schematic/utils/general.py

```diff
@@ -129,40 +129,19 @@ def calculate_datetime(
     return date_time_result
 
 
-def check_synapse_cache_size(
-    directory: str = "/root/.synapseCache",
-) -> float:
-    """use du --sh command to calculate size of .synapseCache.
+def check_synapse_cache_size(directory: str = "/root/.synapseCache") -> float:
+    """Calculate size of .synapseCache directory in bytes using pathlib.
 
     Args:
         directory (str, optional): .synapseCache directory. Defaults to '/root/.synapseCache'
 
     Returns:
-        float: returns size of .synapsecache directory in bytes
+        float: size of .synapsecache directory in bytes
     """
-    # Note: this command might fail on windows user.
-    # But since this command is primarily for running on AWS, it is fine.
-    command = ["du", "-sh", directory]
-    output = subprocess.run(command, capture_output=True, check=False).stdout.decode(
-        "utf-8"
+    total_size = sum(
+        f.stat().st_size for f in Path(directory).rglob("*") if f.is_file()
```

Comment:
There may be some minor discrepancies between how this would operate on windows, mac, and ubuntu

Reply:
This doesn't work on ubunutu:

```
Traceback (most recent call last):
  File "/home/bfauble/BryansGreatWorkspace/schematic/junk.py", line 7, in <module>
    print(check_synapse_cache_size())
  File "/home/bfauble/BryansGreatWorkspace/schematic/schematic/utils/general.py", line 141, in check_synapse_cache_size
    total_size = sum(
  File "/home/bfauble/BryansGreatWorkspace/schematic/schematic/utils/general.py", line 141, in <genexpr>
    total_size = sum(
  File "/home/bfauble/.pyenv/versions/3.10.14/lib/python3.10/pathlib.py", line 1047, in rglob
    for p in selector.select_from(self):
  File "/home/bfauble/.pyenv/versions/3.10.14/lib/python3.10/pathlib.py", line 405, in select_from
    if not is_dir(parent_path):
  File "/home/bfauble/.pyenv/versions/3.10.14/lib/python3.10/pathlib.py", line 1305, in is_dir
    return S_ISDIR(self.stat().st_mode)
  File "/home/bfauble/.pyenv/versions/3.10.14/lib/python3.10/pathlib.py", line 1097, in stat
    return self._accessor.stat(self, follow_symlinks=follow_symlinks)
PermissionError: [Errno 13] Permission denied: '/root/.synapseCache'
```

```
from schematic.utils.general import check_synapse_cache_size

print(check_synapse_cache_size())
```

Reply:
I guess before it probably wouldn't have either since the default is to use `/root`. It does work though with how it's implemented in code:

```
from schematic.store.synapse import SynapseStorage
from schematic.utils.general import check_synapse_cache_size

synapse_store = SynapseStorage()

print(check_synapse_cache_size(synapse_store.root_synapse_cache))
```